### PR TITLE
feat: add preview vector index client

### PR DESCRIPTION
**Pull request by pgautier404**
This commit adds a preview vector index client to the Web SDK. The client currently only implements data plane operations including create, list, and delete index.

### CODE REVIEW

File: examples/web/vector-index/doc-example-files/doc-examples-web-apis.ts
Comment (anitarua) [inline examples/web/vector-index/doc-example-files/doc-examples-web-apis.ts:30]: created -> deleted

Comment (pgautier404) [reply examples/web/vector-index/doc-example-files/doc-examples-web-apis.ts]: Thank you! Fixed in e6d4e4a.

File: packages/client-sdk-web/src/vector-client-props.ts
Comment (cprice404) [inline packages/client-sdk-web/src/vector-client-props.ts:7]: @pgautier404 I think this config object has a bunch of stuff like retries on it that we may want to handle differently for the Vector client. @malandis and I had this same discussion on his Python PR, and I think we are leaning towards having a separate Configuration object for Vector clients so that they can evolve independently.

We got burned a bit by copy/pasting too much of the config/wrapper stuff from CacheClient to TopicClient and AuthClient early on, and we kinda tried to unwind some of that. e.g. we have a separate config object for Topics now. so I want to make sure we're making a deliberate and forward-compatible choice here.

lmkwyt

File: packages/client-sdk-web/src/index.ts
Comment (cprice404) [inline packages/client-sdk-web/src/index.ts:126]: Michael I believe that you decided we wanted to start namespacing these a bit more in python. Do you have thoughts on what we should do here?

**Review by cprice404 — COMMENTED** [review]
overall looking good. exciting that the tests pass. I had one question/nit about the Configuration object, and also would like Michael to weigh in with a quick API review, especially comparing/contrasting the response namespacing pattern that he did in python vs. what we should do here.

File: packages/client-sdk-web/src/vector-client-props.ts
Comment (pgautier404) [reply packages/client-sdk-web/src/vector-client-props.ts:7]: That works for me-- I can imagine we might want to add distinct config for each type.

Comment (pgautier404) [reply packages/client-sdk-web/src/vector-client-props.ts]: Added in 0cc099d.

File: packages/client-sdk-web/src/config/vector-configuration.ts
Comment (cprice404) [inline packages/client-sdk-web/src/config/vector-configuration.ts:17]: will we need these in nodejs SDK too? or does your PR not wire up node.js SDK yet?

Comment (pgautier404) [reply packages/client-sdk-web/src/config/vector-configuration.ts:17]: I'm not doing anything on the node side in this PR. This is specifically for Web SDK and console.

Comment (cprice404) [reply packages/client-sdk-web/src/config/vector-configuration.ts:17]: do you have a sense of how much extra work it would be? I presumed it would be pretty small, and that we would knock them both out together (in fact I'd kind of prefer that we establish a pattern of always adding stuff to both at the same time and not letting them drift from one another).

not a blocker for this PR but if it seems easy and you have bandwidth maybe you could file a follow-up PR before you split.

**Review by cprice404 — APPROVED** [review]
still would really like @malandis to skim this if possible but otherwise I'm +1

File: packages/core/src/messages/responses/list-indexes.ts
Comment (malandis) [inline packages/core/src/messages/responses/list-indexes.ts:38]: To align with the protos

```suggestion
  public getIndexNames() {
    return this.indexNames;
  }
```

**Review by malandis — COMMENTED** [review]
Looks good just one inconsistency vs the protos

**Comment by cprice404** [comment]
@malandis what about the namespacing for the response objects? Do you have thoughts on how we should handle that here for symmetry with what you did in Python?

@pgautier404 you might want to look at the python PR to see what they are doing there. I think the goal is to stop dumping all the response classes into one big namespace and instead have a separate namespace for Vector vs Cache etc.

**Comment by pgautier404** [comment]
> @malandis what about the namespacing for the response objects? Do you have thoughts on how we should handle that here for symmetry with what you did in Python?
> 
> @pgautier404 you might want to look at the python PR to see what they are doing there. I think the goal is to stop dumping all the response classes into one big namespace and instead have a separate namespace for Vector vs Cache etc.

In the Javascript SDKs, there is basically no namespacing whatsoever of classes we export. Every symbol that is exposed to developers is exposed through the root `index.ts` of each SDK. All clients, configs, responses, loggers, and so on are imported from the root. For example, the imports in `examples/nodejs/cache/advanced.ts` are:

```
import {
  CacheGet,
  ListCaches,
  CreateCache,
  CacheSet,
  CacheDelete,
  CacheClient,
  Configurations,
  MomentoLoggerFactory,
  DefaultMomentoLoggerFactory,
  DefaultMomentoLoggerLevel,
  ExperimentalMetricsCsvMiddleware,
  ExperimentalRequestLoggingMiddleware,
  CredentialProvider,
} from '@gomomento/sdk';
```

I don't really see how we could introduce a namespace for the vector index responses that wasn't inherently confusing to users who are importing every other class from the top level. Not to say there isn't a way to introduce namespaced responses in a UX-friendly way. I am not strong enough with TS any bright ideas here though.

**Comment by cprice404** [comment]
discussed this with Michael; there are a few changes I want to get in here before this gets merged.

**Comment by cprice404** [comment]
@malandis I'm done with this and ready to merge it if you are okay with my changes.

I tried a few things for namespacing the vector responses. I was able to separate them a little more cleanly in the file tree, and simplify the export statements in the index files, but I wasn't able to do what I really wanted to do, which was this:

https://github.com/momentohq/client-sdk-javascript/issues/784

I think that's too risky for a `1.x`. So my current thinking is:

* The control plane responses in this PR now have `Vector` in their names
* Users who want better namespacing can opt to import `vector` from the main index, and then reference these as `vector.*`, but
* The types are also available from the main index for consistency / simplicity.
* The data plane response types (coming in the next few PRs) will all be prefixed with `Vector*`, like the current Cache ones are prefixed with `Cache*`.

LMK if you are okay with that.

**Comment by malandis** [comment]
That's fine with me. My only concern was some of the names can get unwieldly-long, eg `VectorAddItemBatchResponse`, `VectorAddItemBatch.Success`. If that's actually not bad then works for me.

**Comment by cprice404** [comment]
> That's fine with me. My only concern was some of the names can get unwieldy/long, eg `VectorAddItemBatchResponse`, `VectorAddItemBatch.Success`. If that's actually not bad then works for me.

I don't see a better option at the moment and I'm hoping that with autocomplete those won't be too bad. I don't think they are meaningfully longer than the existing names for the Cache response classes.

**Review by malandis — APPROVED** [review]

**Pull request merged**
